web: flag explanation: make more readable

lainisourgod opened the pull request:
* add spacing between separate flag explanations
* make flag name bold
* make explanation property array of Text-s so that flag name and explanation may be styled differently

I'm little new to JSX so I got multiple type errors that I don't know how to resolve, but everything works just fine. 

<img width="1105" alt="CleanShot 2023-02-04 at 20 53 29@2x" src="https://user-images.githubusercontent.com/24773652/216779641-b936f392-82ac-4a1e-912e-38deea77fa69.png">

I want in ExplainProps receive not a `string` array, but array of `Text`s that are styled differently.

notmd commented on website/src/components/Explain.tsx:
```diff
@@ -11,7 +11,8 @@ import {
 interface ExplainProps {
-  explanation: string[];
+  // Each element is an explanation for individual item (e.g. flag)
+  explanation: Array<typeof Text>;
```
Maybe replace this with ReactElement[] would work

fozziethebeat replied:
Yeah, i think this will fix the build issue.

notmd reviewed (commented):
I haven't look the visual yet

notmd commented on website/src/components/Explain.tsx:
```diff
@@ -25,7 +26,10 @@ export const Explain = ({ explanation }: ExplainProps) => {
         <PopoverBody>
           {explanation.map((paragraph, idx) => (
-            <Text key={idx}>{paragraph}</Text>
+            <>
```
```suggestion
            <Fragment key={idx}>
```

lainisourgod replied:
I tried changing `<></>` to `Fragment`, linter doesn't like it (1).
```tsx
          {explanation.map((paragraph, idx) =>
            <Fragment key={idx}>
              {paragraph}
              <br />
            </Fragment>
          )}
```

<img width="1134" alt="CleanShot 2023-02-05 at 17 51 14@2x" src="https://user-images.githubusercontent.com/24773652/216823347-04c8104e-dbe1-42af-9b95-fac5698f5afe.png">

If I remove `<br />`, it now doesn't like the `key` property (2)
<img width="1015" alt="CleanShot 2023-02-05 at 17 53 07@2x" src="https://user-images.githubusercontent.com/24773652/216823488-72e5523e-e1ec-48e9-836f-afeb355b6d3e.png">

If I remove `key` property, everything is fine again.

<img width="527" alt="CleanShot 2023-02-05 at 17 54 09@2x" src="https://user-images.githubusercontent.com/24773652/216823571-d14fef91-47e2-496c-bc07-b0e94d4b1823.png">

I'm not really sure what is going on here.

notmd replied:
hey @lainisourgod, sorry for the bad code suggestion, please try this:
```jsx
<Fragment key={idx}>
  <>
    {paragraph}
    <br />
  </>
</Fragment>
```

notmd replied:
Hmm, look like the TS doesn't complain when use change the type to ReactElement[]

lainisourgod replied:
I'm not really proficient in TS type system, but using ReactElement here feels a hack, but if it's ok for your code base, then fine!

fozziethebeat reviewed (changes requested):
Looks good once the build works

lainisourgod commented:
Also @notmd I rebase my code on top of yesterday's commits and now another type issue appeared out of nowhere.
<img width="1272" alt="CleanShot 2023-02-05 at 17 54 47@2x" src="https://user-images.githubusercontent.com/24773652/216823612-39383400-9b65-4eae-80aa-8bffcff48c48.png">

VSCode suggests to put fake children property, I'm not sure it's a good fix.
<img width="1048" alt="CleanShot 2023-02-05 at 17 55 58@2x" src="https://user-images.githubusercontent.com/24773652/216823679-4cfbaa56-e8dd-404f-98e8-9d54e619ac2f.png">

lainisourgod commented:
Also I forgot to pin the screenshot of how it looks now. 

<img width="427" alt="image" src="https://user-images.githubusercontent.com/24773652/216823756-a9742345-a5e5-415f-a47b-cfe8b0af7eab.png">

notmd commented:
> Also @notmd I rebase my code on top of yesterday's commits and now another type issue appeared out of nowhere. <img alt="CleanShot 2023-02-05 at 17 54 47@2x" width="1272" src="https://user-images.githubusercontent.com/24773652/216823612-39383400-9b65-4eae-80aa-8bffcff48c48.png">
> 
> VSCode suggests to put fake children property, I'm not sure it's a good fix. <img alt="CleanShot 2023-02-05 at 17 55 58@2x" width="1048" src="https://user-images.githubusercontent.com/24773652/216823679-4cfbaa56-e8dd-404f-98e8-9d54e619ac2f.png">

Have you tried restarting VSCode? On my end, it doesn't complain anything about it.

lainisourgod commented:
So now another changed happend during the day and now Explanation API is used in other code, so my changes are conflicting with this usage. I wonder if should change the Explanation API back how it was and just render `<br />` between lines without **bold**, or should I update API usage in `LabelLikertGroup.tsx`.

I should check if rendering label name and description seperately is beneficial in `Likert` case, and if it so, change the usage in `LabelLikertGroup.tsx`

lainisourgod commented:
Hi @notmd! Sorry for not being active on issue last week.

I removed `bold` part of changes "**Wrong language:** Not written ..." so not to make API unnecessary complicated and leaved only line breaks.

Also I noticed templating issues likely introduced by #1349 which I fixed by copying templating realization from this PR: https://github.com/LAION-AI/Open-Assistant/blob/main/website/src/components/Messages/LabelFlagGroup.tsx#L44
![CleanShot 2023-02-16 at 01 59 52@2x](https://user-images.githubusercontent.com/24773652/219181820-17950122-2466-40f2-a3e7-e8d07a1d9d63.png)

olliestanley commented:
@notmd @fozziethebeat will both have to review this and confirm that changes requested have been addressed before it can be merged

notmd reviewed (approved):
Thanks

notmd commented:
@olliestanley I just update the PR, can you please force merge if the CI passes?

olliestanley commented:
> @olliestanley I just update the PR, can you please force merge if the CI passes?

I don't have force merge perms, I think will need Fozzie to approve unless @andreaskoepf has time to force merge
